.github/workflows/build_and_test_midstream_image.yml

```diff
@@ -0,0 +1,51 @@
+# This is a basic workflow to help you get started with Actions
+
+name: "Build and test midstream CI imaga"
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: "Build the midstream image inside Dockerfiles"
+        run: |
+          podman --version
+          cd Dockerfiles 
+          podman build -t midstream_image:latest -f Dockerfile --build-arg OPERATOR_SDK_VERSION=v1.4.0 --build-arg OPERATOR_TEST_PLAYBOOKS_TAG=v1.0.11
+
+      - name: "Run example bundle through midstream_image -> POSITVE test"
+        run: |
+          cd Dockerfiles
+          mkdir output_logs
+          podman run -it -v ./example-bundle:/project/operator-bundle -v ./output_logs:/project/output --security-opt label=disable midstream_image:latest
+          # clean output logs for next step
+          rm -rf output_logs/*
+
+      - name: "Run example bundle through midstream_image --> negative test"
+        continue-on-error: true
+        run: |
+          # cd into Dockerfiles since it resets everystep
+          cd Dockerfiles
+          mkdir empty-example-bundle
+          podman run -it -v ./empty-example-bundle:/project/operator-bundle -v ./output_logs:/project/output --security-opt label=disable midstream_image:latest
+
+      - name: "Fail when previous step does not fail"
+        run: |
+          grep -inr "metadata not found in ../operator-bundle" Dockerfiles/output_logs/
```

Comment:
Checking content is not enough. We have o test whether all expected log files are created.